src_compressible/flux.cpp

```diff
@@ -689,7 +708,12 @@ void calculateFlux(const MultiFab& cons_in, const MultiFab& prim_in,
             fluxx(i,j,k,1) = fluxx(i,j,k,1) - (tauxxp+divxp);
             fluxx(i,j,k,2) = fluxx(i,j,k,2) - tauyxp;
             fluxx(i,j,k,3) = fluxx(i,j,k,3) - tauzxp;
-            fluxx(i,j,k,4) = fluxx(i,j,k,4) - (half*phiflx + kxp*(prim(i,j,k,4)-prim(i-1,j,k,4))/dx[0]);
+
+            // heat flux
+            fluxx(i,j,k,nvars) = fluxx(i,j,k,nvars) - (kxp*(prim(i,j,k,4)-prim(i-1,j,k,4))/dx[0]);
+
+            // viscous heating
+            fluxx(i,j,k,nvars+1) = fluxx(i,j,k,nvars+1) - (half*phiflx);
```

Comment:
need to divide by dx?

Reply:
I don't think so. This is directly computing stress tensor contracted with the velocity. The divergence of this term to increment the conserved variable (energy) is done outside in timeStep.cpp

Reply:
The main branch said
fluxx(i,j,k,4) = fluxx(i,j,k,4) - (half*phiflx + kxp*(prim(i,j,k,4)-prim(i-1,j,k,4)) / dx[0]);
so they are inconsistent - let me see if I agree with you as to what the right thing is...

Reply:
phiflx already has a 1/dx scaling so this new code seems correct